### ISSUE TITLE
Handling password attempts

### ISSUE THREAD

**Issue by fivetwelve** [toitoi-co/toitoi-admin#24]
[Rules around password attempts](https://github.com/toitoi-co/toitoi-cms/blob/master/interface_spec.md#003-business-rules). I assume they should be part of -admin since -cms can't keep track of attempts if site is refreshed.

Also, this leads into issue #25, password reset.

**Comment by joepie91** [comment]
Yeah, this would be handled almost entirely by `-admin` - the only thing `-cms` would have to do, would be handling one or more new error types/properties (when limits have been reached), by:

* Showing the 'forgot password?' notice,
* Showing the CAPTCHA when applicable (and sending along the result to the next login call), and
* Showing 'blocked'  messages

I'm not entirely sure how to implement this yet. My thinking is to at least send along certain properties (`requireCaptcha`, `ipBlocked`, ...) with "invalid password" errors, but this wouldn't work UX-wise if the user were to reload the page. There would have to be a way to retrieve the current "attempts left" state whenever the login page is loaded, before actually making an attempt.

Any suggestions?

**Comment by fivetwelve** [comment]
What about writing something to local storage or even a time-sensitive cookie when a flag or flags are sent back in the unsuccessful response. It would not meant to be fool-proof but just something to help the UX side of things, and it would be deleted upon successful login. While I know it's not proof against cleaning out the browser cache, it should be feasible for most of the expected users.

**Comment by joepie91** [comment]
Yeah, that seems like a reasonable solution. I'll just ensure that the API always returns the current status after an attempt, and leave it up to the frontend to persist this information.

**Comment by joepie91** [comment]
The server-side part is now implemented. It will be up to the client-side code to actually display the widget and send along the response, per reCAPTCHA documentation.
